opensfm/reconstruction.py

```diff
@@ -224,24 +224,39 @@ def back_project(camera, shot, pixel, depth):
     return np.linalg.solve(A, b)
 
 
+def distort(camera, x, y):
+    """ Distorts a point (x, y) using the distortion coefficients of the camera.
+
+    :param camera: The camera.
+    :param x: The x coordinate.
+    :param y: The y coordinate.
+    :return: The distorted x and y coordinates.
+    """
+
+    k1 = camera.get('k1', 0.0)
+    k2 = camera.get('k2', 0.0)
+    r2 = x * x + y * y
+    distortion = 1.0 + r2 * (k1 + k2 * r2)
+
+    x_distort = distortion * x
+    y_distort = distortion * y
+
+    return x_distort, y_distort
+
+
 def reproject(camera, shot, point):
     ''' Reproject 3D point onto image plane given a camera
     '''
     p = rotate(shot['rotation'], point['coordinates'])
     p += shot['translation']
-    xp = p[0] / p[2]
-    yp = p[1] / p[2]
+    xp = camera['focal'] * p[0] / p[2]
+    yp = camera['focal'] * p[1] / p[2]
 
-    l1 = camera.get('k1', 0.0)
-    l2 = camera.get('k2', 0.0)
-    r2 = xp * xp + yp * yp
-    distortion = 1.0 + r2  * (l1 + l2  * r2)
-
-    x_reproject = camera['focal'] * distortion * xp
-    y_reproject = camera['focal'] * distortion * yp
+    x_reproject, y_reproject = distort(camera, xp, yp)
```

Comment:
Is this computing the distortion the same way as before?
It looks like before the radius r used to compute distortion was the "normalized image coordinates" (X/Z, Y/Z).  Now, it is using the "undistorted image coordinates" (focal * X/Z, focal * Y/Z)